tests/Unit/TestHelpers.hpp

```diff
@@ -283,3 +286,51 @@ std::string get_output(const Container& c) {
   os << c;
   return os.str();
 }
+
+/*!
+ * \ingroup TestingFramework
+ * \brief Alternative to CAPTURE that prints more digits.
+ */
+#define CAPTURE_PRECISE(variable)                                    \
+  INFO(std::scientific << std::setprecision(18) << #variable << ": " \
+       << (variable));
+
+/*!
+ * \ingroup TestingFramework
+ * \brief Calculates the derivative of an Invocable at a point x, represented by
+ * an array of doubles
+ * in the domain of map with a sixth-order finite difference method.
+ *
+ * \details Intended for use with CoordinateMaps taking the domain {xi,eta,zeta}
+ * to the range {x,y,z}. This function calculates the derivative of a range
+ * index with respect to a domain index. The range index is passed
+ * explicitly to the function as range_index, while the domain index
+ * is deduced from which element of the array dx is non-zero,
+ *
+ * \requires an array dx with a single non-zero coordinate: this
+ * will be the direction along the domain in which the derivative is taken.
+ */
+template <typename Invocable, size_t VolumeDim>
+double derivative(const Invocable& map, const size_t range_index,
+                  const std::array<double, VolumeDim>& x,
+                  const std::array<double, VolumeDim>& dx) {
+  ASSERT(VolumeDim - 1 ==
+             static_cast<size_t>(std::count(dx.begin(), dx.end(), 0.0)),
+         "Can only take derivative along a single logical axis at a time.");
+  ASSERT(VolumeDim > range_index,
+         "Must pass the correct dimension: (0 for a 1D map, 0 or 1 for a 2D "
+         "map, etc.)");
+  const std::array<double, VolumeDim> x_1ahead = x + dx;
+  const std::array<double, VolumeDim> x_2ahead = x_1ahead + dx;
+  const std::array<double, VolumeDim> x_3ahead = x_2ahead + dx;
+  const std::array<double, VolumeDim> x_1behind = x - dx;
+  const std::array<double, VolumeDim> x_2behind = x_1behind - dx;
+  const std::array<double, VolumeDim> x_3behind = x_2behind - dx;
+  const double delta = *std::max_element(dx.begin(), dx.end());
+  return (1.0 / (60.0 * delta)) * gsl::at(map(x_3ahead),range_index) +
+    (-3.0 / (20.0 * delta)) * gsl::at(map(x_2ahead),range_index) +
+    (0.75 / delta) * gsl::at(map(x_1ahead),range_index) +
+    (-0.75 / delta) * gsl::at(map(x_1behind),range_index) +
+    (3.0 / (20.0 * delta)) * gsl::at(map(x_2behind),range_index) +
+    (-1.0 / (60.0 * delta)) * gsl::at(map(x_3behind),range_index);
```

Comment:
I'd like to see a comment detailing the algorithm being used (it looks like 6th order FD?). Probably doesn't need to be in the Doxygen comment, but it would be nice to say it explicitly somewhere.

Reply:
The doxygen comment says the derivatives implemented with a sixth-order FD.

Reply:
-_- ... so it does, apologies.